src/main/kotlin/graphics/scenery/PeriodicTable.kt

```diff
@@ -0,0 +1,111 @@
+package graphics.scenery
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import graphics.scenery.utils.LazyLogger
+import org.joml.Vector3f
+import java.io.FileReader
+import kotlin.collections.ArrayList
+
+/**
+ * Class to parse the periodic table generously provided by PubChem as a json-file (https://pubchem.ncbi.nlm.nih.gov/periodic-table/).
+ * We store all elements as a [ChemicalElement] and put them into the [elementList].
+ */
+open class PeriodicTable {
+
+    /*
+    data-classes to store the JsonProperties
+    The json file has the following format:
+    {Table: {
+        Columns: {
+            Column: [ "Here are the names of all the parameters" ]
+            }
+        Row: [
+            { Cell: [ "Here is the information about the first element stored" ] },
+            { Cell: [ Here is the information about the second element stored" ] }, ... ]
+        }
+     }
+     */
+    data class ChemCell(@JsonProperty("Cell") val cell: ArrayList<String>)
+    data class ChemColumn(@JsonProperty("Column") val column: ArrayList<String>)
+    data class ChemTable(@JsonProperty("Columns") val columns: ChemColumn,@JsonProperty("Row") val row: ArrayList<ChemCell>)
+    data class PeriodicTableau(@JsonProperty("Table") val table: ChemTable)
+
+    val elementList = ArrayList<ChemicalElement>()
+    private val logger by LazyLogger()
+
+    init {
+        //parsing the json file
+        val mapper = jacksonObjectMapper()
+        val file = FileReader("src/main/resources/graphics/scenery/proteins/PubChemElements_all.json").readText()
```

Comment:
This won't work if things are packaged. Use `this::class.java.getResource(filename)` instead!

Reply:
Also, does that JSON file come with any license attached?

Reply:
Done.

There is no license- they kindly ask for a citation, though (https://pubchemdocs.ncbi.nlm.nih.gov/citation-guidelines). I will include this in the doc.